benchmark/__tests__/util.js

```diff
@@ -17,33 +21,360 @@ export function gasBreakdown(outcome) {
+
+  // Convert the gas object to a table-friendly format
+  const breakdownEntries = Object.entries(gasObject.gasBreakdownForReceipt);
+  const breakdownRows = breakdownEntries.map(([key, value]) => ({
+    Description: key,
+    GasUsed: value,
+  }));
+
+  const data = [
+    {
+      Description: "Gas Used to Convert Transaction to Receipt",
+      GasUsed: gasObject.gasUsedToConvertTransactionToReceipt,
+    },
+    {
+      Description: "Gas Used to Execute Receipt",
+      GasUsed: gasObject.gasUsedToExecuteReceipt,
+    },
+    ...breakdownRows,
+    {
+      Description: "Gas Used to Refund Unused Gas",
+      GasUsed: gasObject.gasUsedToRefundUnusedGas,
+    },
+    { Description: "Total Gas Used", GasUsed: gasObject.totalGasUsed },
+  ];
+
+  // Determine column widths
+  const maxDescriptionLength = Math.max(
+    ...data.map((row) => row.Description.length)
+  );
+
+  const descriptionWidth = maxDescriptionLength + 4;
+  const gasWidth = 15; // Increased width for better formatting
+
+  // Create header and separator lines
+  const header = `| ${"Description".padEnd(
+    descriptionWidth
+  )} | ${"Gas Used".padEnd(gasWidth)} |`;
+  const separator = `|${"-".repeat(descriptionWidth + 2)}|${"-".repeat(
+    gasWidth + 2
+  )}|`;
+
+  // Create rows with separators
+  const rows = data
+    .map((row) => {
+      return `| ${row.Description.padEnd(
+        descriptionWidth
+      )} | ${row.GasUsed.padEnd(gasWidth)} |`;
+    })
+    .join(`\n${separator}\n`);
+
+  // Print the results
+  console.log("");
+  console.log(separator);
+  console.log(header);
+  console.log(separator);
+  console.log(rows);
+  console.log(separator);
 }
 
-export function logGasDetail(r, t) {
-  t.log(
-    "Gas used to convert transaction to receipt: ",
-    formatGas(r.result.transaction_outcome.outcome.gas_burnt)
+/**
+ * Generates a gas usage breakdown object from a test result.
+ *
+ * @param {Object} testResult - The result of a test execution, which includes transaction and receipt outcomes.
+ * @param {boolean} [isMinimal=false] - If true, includes only minimal gas usage data; otherwise, includes additional cross-contract call and refund data.
+ * @returns {Object} An object containing formatted gas usage values, including:
+ *   - gasUsedToConvertTransactionToReceipt: Gas used to convert the transaction to a receipt.
+ *   - gasUsedToExecuteReceipt: Gas used to execute the receipt.
+ *   - gasBreakdownForReceipt: Formatted gas breakdown results for the receipt.
+ *   - gasUsedToExecuteCrossContractCall: Gas used to execute any cross-contract calls (if not minimal).
+ *   - gasUsedToRefundUnusedGasForCrossContractCall: Gas used to refund unused gas for cross-contract calls (if not minimal).
+ *   - gasUsedToRefundUnusedGas: Gas used to refund any unused gas (if not minimal).
+ *   - totalGasUsed: Total gas used including all relevant transactions and receipts.
+ */
+export function generateGasObject(testResult, isMinimal = false) {
```

Comment:
In both `generateGasObject` and `logTestResults`, there seems to be overlapping logic for formatting and breaking down gas usage.
Consider extracting common logic into utility functions to avoid repetition and improve code maintainability.